CurrencyConverterService/RatesRecipient.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Net;
+using CurrencyConverterService.Models;
+using System.IO;
+using Newtonsoft.Json;
+using System.Linq;
+
+namespace CurrencyConverterService.Models
+{
+    class RatesRecipient
+    {
+        Context dataBase;
+        Logging logging;
+        List<Rates> rates;
+        CurrencysRecipient currencys;
+        IEnumerable<Currency> currencysDB;
+        Dictionary<int, int> currencyNumbers;
+        List<string> Name;
+        public RatesRecipient()
+        {
+            this.dataBase = new Context();
+            this.logging = new Logging();
+            this.rates = new List<Rates>();
+            this.currencys = new CurrencysRecipient();
+            this.currencyNumbers = new Dictionary<int, int>();
+            this.Name = new List<string>();
+        }
+
+        public List<Rates> GetRatesFromNBRB()
+        {
+            foreach (var currencyNumber in CurrencyNumbers())
+            {
+                if (LastDate(currencyNumber.Key).ToString("yyyy-M-d") != DateTime.Today.AddDays(+1).ToString("yyyy-M-d"))
+                {
```

Comment:
DateTime objects should be compared without to string translation. So, it's better to construct DateTime from the string if any.